src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1213,8 +1219,10 @@ public SolutionState RemoveAdditionalDocument(DocumentId documentId)
 
             var oldProject = this.GetProjectState(documentId.ProjectId);
             var newProject = oldProject.RemoveAdditionalDocument(documentId);
+            var removedDocumentIdAndFilePath = SpecializedCollections.SingletonEnumerable((documentId, oldProject.GetAdditionalDocumentState(documentId).FilePath));
 
-            return this.ForkProject(newProject);
+            return this.ForkProject(newProject,
+                newFilePathToDocumentIdsMap: CreateFilePathToDocumentIdsMapWithRemovedDocuments(removedDocumentIdAndFilePath));
```

Comment:
Do we have any tests for the GetDocumentIdsWithFilePath today that we can expand to confirm we're correctly adding additional files now?

Reply:
I will add a test.